mods/ra/maps/monster-tank-madness/monster-tank-madness.lua

```diff
@@ -103,13 +103,32 @@ SendAlliedUnits = function()
 	Media.PlaySpeechNotification(player, "ReinforcementsArrived")
 	Utils.Do(AlliedUnits, function(table)
 		Trigger.AfterDelay(table.delay, function()
-			Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+			local units = Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+
+			Utils.Do(units, function(unit)
+				if unit.Type == "e6" then
+					Trigger.OnKilled(unit, LandingPossible)
+				end
+			end)
 		end)
 	end)
 
 	Trigger.AfterDelay(DateTime.Seconds(1), function() InitialUnitsArrived = true end)
 end
 
+LandingPossible = function()
+	if not beachReached then
+		Utils.Do(Map.ActorsInBox(Map.TopLeft, Map.BottomRight), function(a)
+			if a.Owner == player then
+				if a.Type == "spen" or a.Type == "lst" then
+					player.MarkFailedObjective(CrossRiver)
```

Comment:
Yes it does - I was too hasty pushing. 

The reason for the spen check is that the same code is called  when the engineer dies and this was the most DRY option I could think of.